File: projet-ebanking-EbankingDevFront/projet-ebanking-EbankingDevFront/src/app/model/dto/RechargeResDTO.ts
import { CCourantResDTO } from '../dto/CCouurantResDTO';

export class RechargeResDTO {
  id!: number;
  operateur!: string;
  phoneNumber!: string;
  montant!: number;
  createdAt!: Date;
  compte!: CCourantResDTO;
}

File: projet-ebanking-ebankingSecuredDev/projet-ebanking-ebankingSecuredDev/src/main/java/com/ebank/ebanking2/model/dto/RechargeResDTO.java
package com.ebank.ebanking2.model.dto;

import com.ebank.ebanking2.model.entity.CCourant;
import com.fasterxml.jackson.annotation.JsonFormat;
import jakarta.persistence.FetchType;
import jakarta.persistence.ManyToOne;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.SuperBuilder;
import org.springframework.data.annotation.CreatedDate;

import java.time.LocalDateTime;

@Data
@AllArgsConstructor
@NoArgsConstructor
@SuperBuilder
public class RechargeResDTO {
    private Long id;
    private String operateur;
    private String phoneNumber;
    private int montant;
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ss")
    private LocalDateTime createdAt;
    private CCourantResDTO compte;
}
